pandas/plotting/_matplotlib/boxplot.py

```diff
@@ -540,7 +540,9 @@ def boxplot_frame_groupby(
                 ax=ax, column=column, fontsize=fontsize, rot=rot, grid=grid, **kwds
             )
             ax.set_title(pprint_thing(key))
-            ret.loc[key] = d
+            # GH 14701 'key' needs to be converted to text as 'key' is a tuple when
+            # there is more than one group
+            ret.loc[pprint_thing(key)] = d
```

Comment:
Instead of `ret = pd.Series(dtype=object)` above, can you do:

```
data = {}
for (key, group), ax in zip(grouped, axes):
    ...
    data[key] = d
ret = pd.Series(data)
```
In general one should not construct a Series row-by-row (this is very slow), but rather provide all the data upfront. I think this will also allow tuples for key.